demos/http/http_demo_basic_tls/http_demo_basic_tls.c

```diff
@@ -0,0 +1,746 @@
+
+static int32_t transportRecv( HTTPNetworkContext_t * pNetworkContext,
+                              void * pBuffer,
+                              size_t bytesToRecv )
+{
+    int32_t bytesReceived = 0;
+    int pollStatus = -1, bytesAvailableToRead = 0;
+    struct pollfd fileDescriptor =
+    {
+        .events  = POLLIN | POLLPRI,
+        .revents = 0
+    };
+
+    /* Set the file descriptor for poll. */
+    fileDescriptor.fd = SSL_get_fd( pNetworkContext->pSslContext );
+
+    /* Check if there are any pending data available for read. */
+    bytesAvailableToRead = SSL_pending( pNetworkContext->pSslContext );
+
+    /* Poll only if there is no data available yet to read. */
+    if( bytesAvailableToRead <= 0 )
+    {
+        pollStatus = poll( &fileDescriptor, 1, TRANSPORT_SEND_RECV_TIMEOUT_MS );
+    }
+
+    /* SSL read of data. */
+    if( ( pollStatus > 0 ) || ( bytesAvailableToRead > 0 ) )
+    {
+        bytesReceived = ( int32_t ) SSL_read( pNetworkContext->pSslContext, pBuffer, bytesToRecv );
+    }
+    /* Poll timed out. */
+    else if( pollStatus == 0 )
+    {
+        LogInfo( ( "Poll timed out and there is no data to read from the buffer." ) );
+    }
+    else
+    {
+        LogError( ( "Poll returned with status = %d.", pollStatus ) );
+        bytesReceived = -1;
+    }
+
+    return bytesReceived;
+}
+
+/*-----------------------------------------------------------*/
+
+static int sendHttpRequest( const HTTPTransportInterface_t * pTransportInterface,
+                            const char * pHost,
+                            const char * pMethod,
+                            const char * pPath )
+{
+    int returnStatus = EXIT_SUCCESS;
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+
+    /* Initialize the request object. */
+    requestInfo.pHost = pHost;
+    requestInfo.hostLen = strlen( pHost );
+    requestInfo.method = pMethod;
+    requestInfo.methodLen = strlen( pMethod );
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. */
+    requestInfo.flags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
+
+    /* Set the buffer used for storing request headers. */
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders,
+                                                      &requestInfo );
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        /* Initialize the response object. The same buffer used for storing
+         * request headers is reused here. */
+        response.pBuffer = userBuffer;
+        response.bufferLen = USER_BUFFER_LENGTH;
+
+        LogInfo( ( "Sending HTTP %s request to %s%s...",
+                   pMethod, SERVER_HOST, pPath ) );
+        LogInfo( ( "Request Headers:\n%.*s",
```

Comment:
I think an argument for making the request headers and body into debug logs instead is because we are sort of testing that the library is correct instead of printing external factors like the transport status and the server response.

Reply:
If we make these into debug logs, do we change log level to debug or keep it as info? Otherwise, what's the point of leaving logs for request headers/body?

Reply:
We keep log level as info. If the log is at debug level, as desired by the demo user, then these will print in addition to all the info logs.

Reply:
I was trying to meet halfway between Archit's suggestion of deleting these logs and your argument for the usefulness of debugging.